File: LeetCode/301-400/345. 反转字符串中的元音字母.java
class Solution {
    public String reverseVowels(String s) {
 char[] cArr = s.toCharArray();
        int left = 0,right = cArr.length -1;
        while (left <right) {
            while (left < right
                    &&
                    !((cArr[left] == 'a')||(cArr[left] == 'e')||(cArr[left] == 'i')
                    ||(cArr[left] == 'o')||(cArr[left] == 'u')||(cArr[left] == 'A')||(cArr[left] == 'E')||(cArr[left] == 'I')
                    ||(cArr[left] == 'O')||(cArr[left] == 'U'))) {
                left++;
            }
            while (left < right
                    &&
                    !((cArr[right] == 'a')||(cArr[right] == 'e')||(cArr[right] == 'i')
                    ||(cArr[right] == 'o')||(cArr[right] == 'u')||(cArr[right] == 'A')||(cArr[right] == 'E')||(cArr[right] == 'I')
                    ||(cArr[right] == 'O')||(cArr[right] == 'U'))) {
                right--;
            }
            char temp = cArr[left];
            cArr[left] = cArr[right];
            cArr[right] = temp;
            left++;
            right--;
        }
        return new String(cArr);
    }
}

File: LeetCode/301-400/345. 反转字符串中的元音字母.go
func reverseVowels(s string) string {
	n := len(s)
	t := []byte(s)
	left, right := 0, n-1
	for left < right {
		for left < right && s[left] != 'a' && s[left] != 'e' && s[left] != 'i' && s[left] != 'o' && s[left] != 'u' && s[left] != 'A' && s[left] != 'E' && s[left] != 'I' && s[left] != 'O' && s[left] != 'U' {
			left++
		}
		for left < right && s[right] != 'a' && s[right] != 'e' && s[right] != 'i' && s[right] != 'o' && s[right] != 'u' && s[right] != 'A' && s[right] != 'E' && s[right] != 'I' && s[right] != 'O' && s[right] != 'U' {
			right--
		}
		t[left], t[right] = t[right], t[left]
		left++
		right--
	}
	return string(t)
}